hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaUtil.java

```diff
@@ -117,6 +118,19 @@ public static void addNamespaceQuota(final Connection connection, final String n
 
   public static void deleteNamespaceQuota(final Connection connection, final String namespace)
       throws IOException {
+    // Before removing namespace quota , remove quota from the tables inside the namespace
+    // which does not have explicit space quotas defined on them.
+    TableName[] tableNames = QuotaUtil.listTableNamesByNamepsace(connection, namespace);
```

Comment:
This case should never happen. You can't delete a namespace which has tables in it. If we're cleaning up a namespace, we shouldn't have any table-level quotas hanging around.

Please log a `WARN` stating that we found a quota entry that shouldn't exist and that we're going to remove it.